### PR TITLE
chore: removed travis yml and added git action support

### PR DESCRIPTION
**Pull request by ozayr-zaviar**
## Summary
- We remove travis yaml from javascript-sdk and moved all the stages to git action.

NOTE:
Please update [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L11), [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L14) and [this](https://github.com/optimizely/ruby-sdk/blob/21005c3085d670e6b3b9c2993ac194163c146c0d/.github/workflows/ruby.yml#L20) line branch name to master before merging.

## Test plan
All tests should pass

### CODE REVIEW

File: .github/workflows/javascript.yml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    branches: [ master ]
+
+jobs:
+  lint_markdown_files:
+    uses: optimizely/javascript-sdk/.github/workflows/lint_markdown.yml@uzair/gitactions
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:16]: v3

File: .github/workflows/javascript.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    with:
+      FULLSTACK_TEST_REPO: ProdTesting
+    secrets:
+      CI_USER_TOKEN: ${{ secrets.CI_USER_TOKEN }}
+      TRAVIS_COM_TOKEN: ${{ secrets.TRAVIS_COM_TOKEN }}
+
+  crossbrowser_and_umd_unit_tests:
+    runs-on: ubuntu-latest
+    env:
+      BROWSER_STACK_USERNAME: ${{ secrets.BROWSERSTACK_USERNAME }}
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:46]: rename it to BROWSERSTACK_USERNAME

File: .github/workflows/javascript.yml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      BROWSER_STACK_ACCESS_KEY: ${{ secrets.BROWSERSTACK_ACCESS_KEY }}
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:47]: same here.

File: .github/workflows/javascript.yml [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    steps:
+    - uses: actions/checkout@v2
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:49]: update across all checkouts to v3.

File: .github/workflows/javascript.yml [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        npm install
+        npm run test-ci
+
+  unit_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node: [ '8', '9', '10', '12', '14' ]
+    steps:
+    - uses: actions/checkout@v2
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:71]: update

File: .github/workflows/javascript.yml [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        npm install
+        npm run coveralls
+    - name: Coveralls Parallel
+      uses: coverallsapp/github-action@master
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }} 
+        path-to-lcov: ./packages/optimizely-sdk/coverage/lcov.info
+        flag-name: run-${{ matrix.node }}
+        # This is a parallel build so need this
+        parallel: true
```
Comment (msohailhussain) [inline .github/workflows/javascript.yml:90]: what does it mean parallel build?

File: packages/optimizely-sdk/package.json [hunk 7]
```diff
@@ -19,10 +19,7 @@
     "prebuild": "npm run clean",
     "build": "rollup -c && cp dist/index.lite.d.ts dist/optimizely.lite.es.d.ts && cp dist/index.lite.d.ts dist/optimizely.lite.es.min.d.ts && cp dist/index.lite.d.ts dist/optimizely.lite.min.d.ts",
     "build-browser-umd": "rollup -c --config-umd",
-    "precover": "nyc npm test",
-    "cover": "nyc report -r lcov",
-    "precoveralls": "npm run cover",
-    "coveralls": "< coverage/lcov.info coveralls",
+    "coveralls": "nyc --reporter=lcov npm test",
```
Comment (msohailhussain) [inline packages/optimizely-sdk/package.json:22]: why is this removed?

**Review by msohailhussain — CHANGES_REQUESTED** [review]
please address comments and also revert links to master.

File: .github/workflows/javascript.yml (hunk 6 above)
Comment (ozayr-zaviar) [reply .github/workflows/javascript.yml:90]: we are running unit test on multiple node versions and multiple posts on coveralls are required, thats why we are sending those posts in parallel.

File: packages/optimizely-sdk/package.json (hunk 7 above)
Comment (ozayr-zaviar) [reply packages/optimizely-sdk/package.json:22]: The previous `precoveralls` and `cover` is required to post coveralls from travis but it is not working on git actions so we post using 3rd party git action `coverallsapp/github-action@master` and is no longer require here.

**Review by msohailhussain — APPROVED** [review]
lgtm

**Review by jaeopt — APPROVED** [review]
LGTM

File: .github/workflows/integration_test.yml [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        # You should create a personal access token and store it in your repository
+        token: ${{ secrets.CI_USER_TOKEN }}
+        repository: 'optimizely/travisci-tools'
+        path: 'home/runner/travisci-tools'
+        ref: 'master'
+    - name: set SDK Branch if PR
+      if: ${{ github.event_name == 'pull_request' }}
+      run: |
+        echo "SDK_BRANCH=${{ github.head_ref }}" >> $GITHUB_ENV
+        echo "TRAVIS_BRANCH=${{ github.head_ref }}" >> $GITHUB_ENV
```
Comment (msohailhussain) [inline .github/workflows/integration_test.yml:29]: remove it from all places.

**Pull request merged**